src/main/java/org/researchspace/services/storage/file/NonVersionedFileStorage.java

```diff
@@ -40,31 +41,62 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.eclipse.rdf4j.model.IRI;
 import org.researchspace.services.storage.StorageUtils;
 import org.researchspace.services.storage.api.*;
+import org.researchspace.services.storage.api.PathMapping;
 
 import com.google.common.collect.ImmutableList;
 
 public class NonVersionedFileStorage implements ObjectStorage {
-    public final static String STORAGE_TYPE = "nonVersionedFile";
+    public static final String STORAGE_TYPE = "nonVersionedFile";
     private static final String FIXED_REVISION = "";
-    
+
     private static final Logger logger = LogManager.getLogger(NonVersionedFileStorage.class);
 
     private final PathMapping paths;
     private final Config config;
 
     public NonVersionedFileStorage(PathMapping paths, Config config) {
-        this.paths = paths;
+
         this.config = config;
+
+        // Check if the config requires a hierarchical path mapping
+        String pathMapping = config.getPathMapping();
+        IRI pathMappingBaseIri = config.getPathMappingBaseIri();
+
+        if (Objects.nonNull(pathMapping) && Objects.nonNull(pathMappingBaseIri) && pathMapping.equals("hierarchical")) {
```

Comment:
I believe it is a wrong place for that, probably a better one is https://github.com/researchspace/researchspace/blob/3b072f67f245e77421a87842d1c02afe5652c45d/src/main/java/org/researchspace/services/storage/MainPlatformStorage.java#L180 . So if you move this logic there and also move pathMapping and pathMappingBaseIri into StorageConfig class we can reuse the same logic in the S3Storage and in the GitStorage without any additional effort

Reply:
The last commit has been moved to a new proper branch: https://github.com/researchspace/researchspace/tree/hierarchical.